core/src/main/java/org/apache/calcite/sql/SqlSelect.java

```diff
@@ -127,6 +127,9 @@ public SqlOperator getOperator() {
     case 9:
       fetch = operand;
       break;
+    case 10:
+      hints = (SqlNodeList) operand;
+      break;
```

Comment:
Thanks for fixing this, can we try to give a test case ? I know this is a bug caused by me, but i still expect a test.

Reply:
ok, i will submit a test in a minute